plugins/woocommerce/includes/class-wc-order-factory.php

```diff
@@ -28,13 +31,26 @@ public static function get_order( $order_id = false ) {
 			return false;
 		}
 
+		$use_orders_cache = OrderUtil::should_use_orders_cache();
+		if ( $use_orders_cache ) {
+			$order_cache = wc_get_container()->get( OrderCache::class );
+			$order       = $order_cache->get( $order_id );
+			if ( ! is_null( $order ) ) {
```

Comment:
let's also add a `instance of` check, to accommodate for cases where cache engine has some issue and is returning an error object.

Reply:
Or perhaps we should add them in the `ObjectCache.php` itself to address similar cases everywhere.

Reply:
> to accommodate for cases where cache engine has some issue and is returning an error object

This can't happen: the cache engine must either return a cached object or null (that's how the `CacheEngine` interface is defined), issues can be logged but the cache engine classes should never throw exceptions or return error objects.